### ISSUE TITLE
Discuss impact of GET /wallet/transactions ordering request

### ISSUE THREAD

**Issue by placer14** [OpenBazaar/openbazaar-desktop#1558]
Brief Description: @drwasho wants this response to be chronologically ordered within the `transactions` key. JSON should be able to maintain this ordering and the backend has the data to provide it. Need to determine impact for desktop UI. (https://github.com/OpenBazaar/openbazaar-go/issues/1106#issuecomment-419272334)

I expect the only change is that requests to this endpoint will be chronologically ordered instead of transactional (persistence) order. These may be the same when requesting a specific coin, but expect when not specifying a coin that all coins would be collated together.

Would like to know that if we apply this change, that desktop is tracking it and has no problem with the changed scope.

**Comment by jjeffryes** [comment]
I don't think we expect any particular order now, so this shouldn't have an impact. 

They transactions come with a timestamp already, it would be safer to use that for ordering within the mobile client than to rely on just the order in the JSON response.

@rmisio and @jashot7 what do you think?

**Comment by placer14** [comment]
BTW, I agree the mobile client should not depend on this ordering. That said, we're already doing a lot of data "prep" before transmitting data over the wire, so reordering the data before transit is not much of an extra burden.

**Comment by rmisio** [comment]
> I don't think we expect any particular order now, so this shouldn't have an impact.

I don't think that's true. We're expecting it in reverse chronological order, which is the way it's coming. How is what @drwasho wants different? If it is different, it would be a problem because I believe the UX req (at least on desktop and I assume mobile too) is for those to display in reverse chronological order. Wouldn't you want the most recent transactions at the top of the list?

> They transactions come with a timestamp already, it would be safer to use that for ordering within the mobile client than to rely on just the order in the JSON response.

That doesn't work if you are fetching the transactions in a paginated way (using the limit and offsetId params). For pagination it is important the server sorts them in the way you want.

If Washo really needs a different sort order, I would suggest making a sort parameter where different clients can control what field and what direction (ascending/descending) the result set is sorted by. But... tbh I'm not sure how that would work with the offsetId. The offsetId works under the premise that the data set is sorted in insertion or reverse insertion order.

**Comment by placer14** [comment]
He's specifically talking about when no coin type is specified, he would like ALL transactions to be collated into chronological order. I guess the existing implementation just dumps each transaction set from each wallet sequentially instead of collating them into a single set? (I'm just guessing, I haven't tested this endpoint on the `multiwallet` branch.) Also, he didn't specify the direction of the order... I think reverse chronological (most recent first) is what he intends. @drwasho Please clarify if otherwise.

**Comment by rmisio** [comment]
> I guess the existing implementation just dumps each transaction set from each wallet sequentially instead of collating them into a single set?

If that's the case, that really sucks! I would think and hope the query could be made in such a way where it just selects all from the db, which by default would get you reverse chronological order.

Or, are the transactions for different coins in different tables? If so, I'm not sure how you would paginate in that case and IMHO not being able to paginate is much less than ideal, especially for remote servers having to pull a potentially huge result set. (I guess that's not a problem for mobile and desktop isn't making the call, so we could bury our head in the sand on this and hope a requirement for the ideal way doesn't crop up at an inopportune time).

@cpacia Are the transactions for each coin in their own tables?

**Comment by placer14** [comment]
@rmisio We manipulate the transactions in memory and don't expect them to come out of storage collated. This allows us to do the pagination required and shouldn't be a problem.

**Comment by jjeffryes** [comment]
@placer14 What's the status of this on the backend?
